var/spack/repos/builtin/packages/llvm/package.py

```diff
@@ -250,17 +255,25 @@ def cmake_args(self):
             '-DLLVM_ENABLE_EH:BOOL=ON',
             '-DCLANG_DEFAULT_OPENMP_RUNTIME:STRING=libomp',
             '-DPYTHON_EXECUTABLE:PATH={0}'.format(spec['python'].command.path),
+            '-DLIBOMP_USE_HWLOC=On',
         ]
 
         projects = []
 
-        # TODO: Instead of unconditionally disabling CUDA, add a "cuda" variant
-        #       (see TODO above), and set the paths if enabled.
-        cmake_args.extend([
-            '-DCUDA_TOOLKIT_ROOT_DIR:PATH=IGNORE',
-            '-DCUDA_SDK_ROOT_DIR:PATH=IGNORE',
-            '-DCUDA_NVCC_EXECUTABLE:FILEPATH=IGNORE',
-            '-DLIBOMPTARGET_DEP_CUDA_DRIVER_LIBRARIES:STRING=IGNORE'])
+        if '+cuda' in spec:
+            cmake_args.extend([
+                '-DCUDA_TOOLKIT_ROOT_DIR:PATH=' + spec['cuda'].prefix,
+                '-DCUDA_NVCC_EXECUTABLE:FILEPATH=' + spec['cuda'].prefix.bin + '/nvcc',
+                '-DLIBOMPTARGET_NVPTX_COMPUTE_CAPABILITIES:LIST={0}'.format(','.join(spec.variants['nvptx_offload_ccs'].value)),
+                '-DCLANG_OPENMP_NVPTX_DEFAULT_ARCH=sm_{0}'.format(spec.variants['nvptx_offload_ccs'].value[-1]),
+            ])
+        else:
+            # still build libomptarget but disable cuda
+            cmake_args.extend([
+                '-DCUDA_TOOLKIT_ROOT_DIR:PATH=IGNORE',
+                '-DCUDA_SDK_ROOT_DIR:PATH=IGNORE',
+                '-DCUDA_NVCC_EXECUTABLE:FILEPATH=IGNORE',
```

Comment:
Same comment here, I think it suffices to just have `-DCUDA_TOOLKIT_ROOT_DIR=...`